metadata-ingestion/src/datahub/ingestion/source/redshift/lineage.py

```diff
@@ -161,6 +167,73 @@ def __init__(
             self.report.lineage_end_time,
         ) = self.get_time_window()
 
+    def _init_temp_table_schema(
+        self, database: str, temp_tables: List[TempTableRow]
+    ) -> None:
+        if self.context.graph is None:  # to silent lint
+            return
+
+        schema_resolver: sqlglot_l.SchemaResolver = (
+            self.context.graph._make_schema_resolver(
+                platform=LineageDatasetPlatform.REDSHIFT.value,
+                platform_instance=self.config.platform_instance,
+                env=self.config.env,
+            )
+        )
+
+        dataset_vs_columns: Dict[str, List[SchemaField]] = {}
+        # prepare dataset_urn vs List of schema fields
+        for table in temp_tables:
+            result = sqlglot_l.create_lineage_sql_parsed_result(
+                platform=LineageDatasetPlatform.REDSHIFT.value,
+                platform_instance=self.config.platform_instance,
+                env=self.config.env,
+                database=database,
+                schema=self.config.default_schema,
+                query=table.query_text,
+                graph=self.context.graph,
+            )
+
+            if result.column_lineage is None:
```

Comment:
Here, you should filter for results for `CREATE` statements only.
Please, make sure it won't filter out CTAS (create table as select statements)

Reply:
Confirmed, It is not filtering out the CTAS